core/src/main/java/org/apache/brooklyn/policy/InvokeEffectorOnSensorChange.java

```diff
@@ -79,18 +79,26 @@ public void setEntity(EntityLocal entity) {
             LOG.debug("Defaulting to producer==self for {}, on entity {}", this, entity);
             producer = entity;
         }
+        lastValue = entity.sensors().get(sensor);
         subscriptions().subscribe(producer, sensor, this);
         LOG.debug("{} subscribed to {} events on {}", new Object[]{this, sensor, entity});
     }
 
     @Override
     public void onEvent(SensorEvent<Object> event) {
-        final Effector<?> eff = getEffectorNamed(getConfig(EFFECTOR)).get();
+        final Object newValue = event.getValue();
+        LOG.debug("lastValue='{}', newValue='{}'", lastValue, newValue);
+        if (String.valueOf(newValue).equals(lastValue)) {
```

Comment:
I don't understand why you're converting to a string here - sensors can hold any object type.

Reply:
- [ ] You are right, I'll cast this to `Comparable`, left this from experimentation.

Reply:
9cfe27fdca482df87febc49aa6a8b071512a44fa